### ISSUE TITLE
Default font is oblique on Mac OS X

### ISSUE THREAD

**Issue by RivetBenoit** [pyzo/pyzo#461]
The default font for Pyzo is set to DejaVu Sans Mono; however on my computer, running Mac OS Sierra (Mac OS X.12), it's the oblique shape which is used by default; there is no way to choose the regular shape instead and this is pretty annoying in my opinion. A quick workaround is to remove the Oblique versions of the DejaVu Sans Mono fonts in the pyzo.app/pyzo/resource/fonts folder; but it would be nicer if the default version chosen by Pyzo could be the normal shape and not the oblique one, or if we could choose between the regular version and the oblique version of the font.

In the dialog boxes : Settings->Edit syntax styles or Settings->Advanced settings don't do anything but pop a dialog box explaining that the corresponding actions are not available but that we are free to change it in the appropriate files. This is annoying, but since it's always been the case, any old user of Pyzo knows it would be useless to hold one's breath until it's been implemented. However, the dialog box reads : "In this 3.0 release, ...", which is odd when opening the 4.3.1 release of Pyzo. It should possible to track all the occurrences of the version number in the Pyzo code and update them :-)

Thanks to Almar and all the people maintaining Pyzo for your hard work : this is really appreciated :-)

**Comment by almarklein** [comment]
Thanks for the feedback!

About the version number: I change the text in this dialog to avoid confusion.

About the font: this is really odd. Does there happen to be a "Could not load font" message in the logger shell? I just made a change to load the fonts in order (the non-oblique first). With a bit of luck, this helps. Otherwise I am not sure where to look for a solution.

**Comment by almarklein** [comment]
I now have a Macbook to test things like this on. And I can reproduce this bug. @rreilink do you see this too with the latest release (or latest master)?

**Comment by almarklein** [comment]
I looked into this but not a lot of progress so far:

* I can reproduce it with the latest Pyzo binaries (4.4.0, which are build with PySide).
* When running from source with Miniconda and PyQt5 provided by Continuum, I *do not* see the error.
* When running from source with Miniconda and PySide provided by conda-forge, I *do* see the error.
* Cannot test with PyQt4 at the moment because of unsatisfyable constraints of conda packages.
* I tried updating the font files to the latest version [2.37](https://dejavu-fonts.github.io/Download.html). No luck.
* I checked the code in `menu.py` and `setFont()` in `codeeditor/base.py`, but everything seems fine.
* I can confirm that removing the oblique font file fixes the issue.

This is starting to look like an issue specific to PySide, or perhaps Qt4, related to how Qt selects the font. When the oblique font is removed, the editor can still show text in italic, probably because Qt does the slanting itself. So it looks like we can just remove the file from the repo.

Any other thoughs or ideas?
